gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestPlanMapperTest.java

```diff
@@ -551,6 +551,160 @@ public void When_MappingDiscontinue_With_MissingPertinentInformation_Expect_Defa
         );
     }
 
+    @Test void When_MappingDiscontinue_With_CodingOriginalTextIsDuplicatedAsPrefixInPertinentInformation_Expect_DisplayedOnce() {
+        var medicationStatementXml = """
+            <MedicationStatement xmlns="urn:hl7-org:v3" classCode="SBADM" moodCode="INT">
+                <id root="B4D70A6D-2EE4-41B6-B1FB-F9F0AD84C503"/>
+                <component typeCode="COMP">
+                    <ehrSupplyAuthorise classCode="SPLY" moodCode="INT">
+                        <id root="TEST_ID"/>
+                    </ehrSupplyAuthorise>
+                </component>
+                <component typeCode="COMP">
+                    <ehrSupplyDiscontinue classCode="SPLY" moodCode="RQO">
+                        <id root="D0BF39CA-E656-4322-879F-83EE6E688053"/>
+                        <code code="EMISDRUG_DISCONTINUATION" codeSystem="2.16.840.1.113883.2.1.6.3">
+                            <originalText>Prescribing error</originalText>
+                        </code>
+                        <availabilityTime value="20060426"/>
+                        <reversalOf typeCode="REV">
+                            <priorMedicationRef classCode="SBADM" moodCode="ORD">
+                                <id root="TEST_ID"/>
+                            </priorMedicationRef>
+                        </reversalOf>
+                        <pertinentInformation typeCode="PERT">
+                            <pertinentSupplyAnnotation classCode="OBS" moodCode="EVN">
+                                <text>Prescribing error, incorrect dosage</text>
+                            </pertinentSupplyAnnotation>
+                        </pertinentInformation>
+                    </ehrSupplyDiscontinue>
+                </component>
+            </MedicationStatement>
+            """;
+        var ehrExtract = unmarshallEhrExtractFromMedicationRequestXml(medicationStatementXml);
+        var medicationStatement = extractMedicationStatement(ehrExtract);
+        var supplyAuthorise = extractSupplyAuthorise(medicationStatement);
+
+        var medicationRequest = medicationRequestPlanMapper.mapToPlanMedicationRequest(
+                ehrExtract,
+                medicationStatement,
+                supplyAuthorise,
+                PRACTISE_CODE
+        );
+        var statusExt = medicationRequest.getExtensionsByUrl(MEDICATION_STATUS_REASON_URL);
+        var statusReasonExt = statusExt.get(0).getExtensionsByUrl(STATUS_REASON);
+        var statusReason = (CodeableConcept) statusReasonExt.get(0).getValue();
+
+        assertAll(
+            () -> assertThat(statusExt).hasSize(1),
+            () -> assertThat(statusReasonExt).hasSize(1),
+            () -> assertThat(statusReason.getText()).isEqualTo("Prescribing error, incorrect dosage")
+        );
+    }
+
+    @Test void When_MappingDiscontinue_With_CodingOriginalTextIsDuplicatedAsUpperCasePrefixInPertinentInformation_Expect_DisplayedTwice() {
```

Comment:
The tests are very similar, so they can be converted into the parameterized ones which should save a lot of code lines and make it more concise...